xarray/tests/test_backends.py

```diff
@@ -1030,17 +1105,19 @@ def test_open_mfdataset(self):
             with create_tmp_file() as tmp2:
                 original.isel(x=slice(5)).to_netcdf(tmp1)
                 original.isel(x=slice(5, 10)).to_netcdf(tmp2)
-                with open_mfdataset([tmp1, tmp2]) as actual:
-                    self.assertIsInstance(actual.foo.variable.data, da.Array)
-                    self.assertEqual(actual.foo.variable.data.chunks,
-                                     ((5, 5),))
-                    self.assertDatasetAllClose(original, actual)
-                with open_mfdataset([tmp1, tmp2], chunks={'x': 3}) as actual:
-                    self.assertEqual(actual.foo.variable.data.chunks,
-                                     ((3, 2, 3, 2),))
+                for close in [True, False]:
+                    with open_mfdataset([tmp1, tmp2], autoclose=close) as actual:
+                        self.assertIsInstance(actual.foo.variable.data, da.Array)
+                        self.assertEqual(actual.foo.variable.data.chunks,
+                                         ((5, 5),))
+                        self.assertDatasetAllClose(original, actual)
+                    with open_mfdataset([tmp1, tmp2], chunks={'x': 3}, autoclose=close) as actual:
+                        self.assertEqual(actual.foo.variable.data.chunks,
+                                         ((3, 2, 3, 2),))
 
         with self.assertRaisesRegexp(IOError, 'no files to open'):
-            open_mfdataset('foo-bar-baz-*.nc')
+            for close in [True, False]:
```

Comment:
The loop needs to be outside the `with self.assertRaisesRegexp` clause.

But more generally, let's think a little carefully about whether we really need to test all of these cases for both autoclose=True and autoclose=False. My guess is that this is more than is necessary.

I would also prefer to use `pytest` style test fixtures for the `autoclose` argument. This ensures that each option ends up in a different test, which makes things much easier to debug when we get a failure.